lib/semantics/check-omp-structure.cc

```diff
@@ -14,9 +14,82 @@
 
 #include "check-omp-structure.h"
 #include "../parser/parse-tree.h"
+#include <variant>
 
 namespace Fortran::semantics {
 
+bool OmpStructureChecker::HasInvalidCloselyNestedRegion(
+    const parser::CharBlock &source, const OmpDirectiveEnumType &set) {
+  if (ompContext_.size() && set.test(GetOmpContext().currentDirectiveEnum)) {
```

Comment:
Call to `size()` is to make sure that the given directive (represented here as `source`) is at least nested within one level of other directives (at least one `OmpContext` is on the stack). Otherwise (`0`) the function just returns `false`.

Reply:
Addressed in commit [cb89744](cb89744746186bcf120d8ca62a791e2778366b9b). A small member function is used instead to check whether a directive is nested in other regions.